src/metabase/pulse/render.clj

```diff
@@ -147,11 +180,12 @@
       (if (:card-error (ex-data e))
         (do
           (log/error e "Pulse card query error")
-          (body/render :card-error nil nil nil nil nil))
+          (assoc (body/render :card-error nil nil nil nil nil)
+                 :attachments (error-file-attachment card e)))
         (do
           (log/error e "Pulse card render error")
-          (body/render :render-error nil nil nil nil nil))))))
-
+          (assoc (body/render :render-error nil nil nil nil nil)
```

Comment:
wdyt instead of manually creating an attachment here, we make that an implementation detail of `body/render :render-error`?

we could pass the exception as data and let `body/render` creates an attachment for it.